nilearn/surface/surface.py

```diff
@@ -894,10 +907,9 @@ def _check_mesh(mesh):
 def check_mesh_and_data(mesh, data):
```

Comment:
We never check that the faces are consistent with coordinates, e.g. faces.max() <= len(coords)

Reply:
Indeed! This should be addressed in 088f2a4